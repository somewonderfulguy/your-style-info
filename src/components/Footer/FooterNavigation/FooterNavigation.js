import React from 'react'

import LinkExtended from 'components/LinkExtended'
import {PRIME_ROUTES} from 'constants/index'
import {useLocalisation} from 'contexts'
import styles from './FooterNavigation.module.css'

const FooterNavigation = () => {
  const {locale} = useLocalisation()

  return (
    <nav>
      <ul className={styles.list}>
        {Object.entries(PRIME_ROUTES) && Object.entries(PRIME_ROUTES).map(([path, {name, inactive}]) => (
          <li key={path}>
            {inactive ? (
              <span className={styles.inactive} aria-disabled>{name}</span>
            ) : (
              <LinkExtended to={`/${locale}${path}`} className={styles.link} activeClassName={styles.active}>
                {name}
              </LinkExtended>
            )}
          </li>
        ))}
      </ul>
    </nav>
  )
}

export default FooterNavigation