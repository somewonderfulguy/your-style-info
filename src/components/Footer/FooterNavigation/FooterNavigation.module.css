.link {
  color: #fff;
}

.active {
  color: lightgray;
  font-weight: 600;
}