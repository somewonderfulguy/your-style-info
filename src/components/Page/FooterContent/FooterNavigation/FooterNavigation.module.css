.list {
  list-style-type: none;
  display: grid;
  grid-template-rows: repeat(4, min-content);
  grid-auto-flow: column;
  grid-auto-columns: 250px;
  gap: 10px;
}

@media (max-width: 425px) {
  .list {
    grid-template-rows: initial;
    grid-auto-flow: initial;
    grid-auto-columns: auto;
  }
}

.link {
  color: #919191;
}

.link:hover {
  color: lightgray;
}

.active {
  color: lightgray;
  font-weight: 600;
}

.inactive {
  color: #4a4a4a;
}
