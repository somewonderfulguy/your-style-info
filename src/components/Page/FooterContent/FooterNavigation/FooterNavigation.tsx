import LinkExtended from '~components/LinkExtended'
import { PRIME_ROUTES } from '~constants/index'
import { useLocalization } from '~contexts/localizationContext'

import styles from './FooterNavigation.module.css'

const FooterNavigation = () => {
  const [{ locale }] = useLocalization()

  return (
    <nav>
      <ul className={styles.list}>
        {Object.entries(PRIME_ROUTES) &&
          Object.entries(PRIME_ROUTES).map(([path, { name, inactive }]) => (
            <li key={path}>
              {inactive ? (
                <span className={styles.inactive} aria-disabled>
                  {name}
                </span>
              ) : (
                <LinkExtended
                  to={`/${locale}${path}`}
                  className={styles.link}
                  activeClassName={styles.active}
                >
                  {name}
                </LinkExtended>
              )}
            </li>
          ))}
      </ul>
    </nav>
  )
}

export default FooterNavigation
