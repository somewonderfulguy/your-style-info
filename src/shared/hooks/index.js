export * from './useForceUpdate'
export * from './useOutsideClick'
export * from './usePrevious'
export * from './useResizeObserver'
export * from './useWindowResize'