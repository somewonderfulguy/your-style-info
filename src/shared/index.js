export * from './componentRenderer'
export * from './getComponent'
export * from './tests'