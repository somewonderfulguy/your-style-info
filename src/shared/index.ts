export * from './hooks'
export * from './tests'
export * from './utils'

export * from './types'