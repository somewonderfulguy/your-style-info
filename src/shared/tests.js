import React from 'react'
import {render as rtlRender} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {Router} from 'react-router-dom'
import {createMemoryHistory} from 'history'
import {QueryClient, QueryClientProvider} from 'react-query'

import {HeaderHeightProvider, LocalizationProvider, ScreenDimensionsProvider, ThemeProvider} from 'contexts'

const queryClient = new QueryClient({
  defaultOptions: {
    initialStale: true
  }
})

const render = (ui, {route = '/', ...options} = {}) => {
  const history = createMemoryHistory({initialEntries: [route]})
  const Wrapper = props => (
    <QueryClientProvider client={queryClient}>
      <Router history={history}>
        <HeaderHeightProvider>
          <LocalizationProvider>
            <ThemeProvider>
              <ScreenDimensionsProvider {...props} />
            </ThemeProvider>
          </LocalizationProvider>
        </HeaderHeightProvider>
      </Router>
    </QueryClientProvider>
  )
  return {
    ...rtlRender(ui, {wrapper: Wrapper, ...options}),
    history,
  }
}

const renderWithoutProviders = rtlRender

export * from '@testing-library/react'
export {render, renderWithoutProviders, userEvent}