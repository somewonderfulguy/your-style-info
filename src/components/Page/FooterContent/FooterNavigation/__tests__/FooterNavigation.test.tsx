import React, {ReactNode} from 'react'

import {render, screen} from 'shared/tests'
import FooterNavigation from '..'

// eslint-disable-next-line react/display-name
jest.mock('components/LinkExtended', () => ({to, children}: {to: string, children: ReactNode}) => <a href={to}>{children}</a>)

test('should match snapshot', () => {
  const {asFragment} = render(<FooterNavigation />)
  expect(asFragment()).toMatchSnapshot()
})

test('should render menu items as links and inactive items as usual text', () => {
  render(<FooterNavigation />)

  const listItems = screen.getAllByRole('listitem')
  const links = screen.getAllByRole('link')
  const inactiveItems = screen.getByRole('list').querySelectorAll('*[aria-disabled="true"]')

  expect(listItems).toHaveLength(7)
  expect(links).toHaveLength(6)
  expect(inactiveItems).toHaveLength(1)
})