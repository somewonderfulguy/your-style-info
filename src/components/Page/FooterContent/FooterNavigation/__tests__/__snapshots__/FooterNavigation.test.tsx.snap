// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="list"
    >
      <li>
        <a
          href="/en/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/en/guides-topics"
        >
          Guides & generic topics
        </a>
      </li>
      <li>
        <a
          href="/en/clothes"
        >
          Clothes & Accessories
        </a>
      </li>
      <li>
        <a
          href="/en/grooming"
        >
          Grooming
        </a>
      </li>
      <li>
        <a
          href="/en/brands-stores"
        >
          Brands & stores
        </a>
      </li>
      <li>
        <a
          href="/en/glossary"
        >
          Glossary
        </a>
      </li>
      <li>
        <span
          aria-disabled="true"
          class="inactive"
        >
          Mixed
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
